### PR TITLE
Puppetv4 and random improvements

### PR DESCRIPTION
**Pull request by anders-larsson**

### CODE REVIEW

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 require 'spec_helper'
 describe 'sudo' do
   context 'with class default options' do
+    let :facts do
+      {
+        :architecture => 'x86_64',
+      }
+    end
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: Wouldn't it make sense to define default facts for the spec tests and override them only when needed ?
See https://github.com/ghoneycutt/puppet-module-nrpe/commit/bf58520ef7fcab0a1f57da7f65e81c2feddaf435 for an example.

File: Gemfile [hunk 2]
```diff
@@ -6,11 +6,16 @@ else
...
+if RUBY_VERSION <= '1.8.7'
+  gem 'rake', '~> 10.1'
+end
+
 gem 'metadata-json-lint'
 gem 'puppetlabs_spec_helper', '>= 0.1.0'
 gem 'puppet-lint', '>= 1.0.0'
 gem 'facter', '>= 1.7.0'
-gem 'rspec-puppet', '~>1.0'
+gem 'rspec-puppet', '~> 2.0'
```
Comment (ghoneycutt) [inline Gemfile]: I don't think this is needed as I don't pin rspec-puppet

https://github.com/ghoneycutt/puppet-module-nrpe/blob/master/Gemfile

Comment (anders-larsson) [inline Gemfile]: Ok. I did it without limiting it to a specific version at first but then looked at puppetlabs/puppetdb and they seemed to have limited it to 2.0.

Update incoming.

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (anders-larsson) [inline spec/classes/init_spec.rb]: Ok. Fix incoming!

File: .travis.yml [hunk 3]
```diff
@@ -34,6 +36,14 @@ matrix:
       env: PUPPET_GEM_VERSION="~> 3.3.0"
     - rvm: 2.1.0
       env: PUPPET_GEM_VERSION="~> 3.4.0"
+    - rvm: 1.8.7
+      env: PUPPET_GEM_VERSION="~> 4.2.0"
+    - rvm: 1.8.7
+      env: PUPPET_GEM_VERSION="~> 4.3.0"
```
Comment (Phil-Friderici) [inline .travis.yml]: please add explicit support for Puppet v4.4 like this:
<pre>
    - rvm: 1.8.7
      env: PUPPET_GEM_VERSION="~> 4.4.0"
</pre>

File: .travis.yml [hunk 4]
```diff
@@ -16,6 +16,8 @@ env:
     - PUPPET_GEM_VERSION="~> 3.5.1"
     - PUPPET_GEM_VERSION="~> 3.6.0"
     - PUPPET_GEM_VERSION="~> 3.7.0"
```
Comment (Phil-Friderici) [inline .travis.yml]: oh! Travis checks are quite outdated. Please add these lines here too:
<pre>
    - PUPPET_GEM_VERSION="~> 3.8.0"
    - PUPPET_GEM_VERSION="~> 3" FUTURE_PARSER="yes"
    - PUPPET_GEM_VERSION="~> 4.0.0"
    - PUPPET_GEM_VERSION="~> 4.1.0"
    - PUPPET_GEM_VERSION="~> 4.2.0"
    - PUPPET_GEM_VERSION="~> 4.3.0"
    - PUPPET_GEM_VERSION="~> 4.4.0"
</pre>

In the end .travis.yml should look close too https://github.com/Phil-Friderici/puppet-module-afs/blob/master/.travis.yml

File: .travis.yml (hunk 3 above)
Comment (anders-larsson) [inline .travis.yml]: Found that I was actually only testing 4 and 4 STRICT_VARIABLES="yes". I've added 4.0.0 to 4.4.0 to the tests now.

Comment (Phil-Friderici) [inline .travis.yml]: Thanks Anders !

**Comment by ghoneycutt** [comment]
Since you are adding puppetv4 support, suggest updating metadata and .fixtures.yml to use stdlib 4.6.0 and update the metadata to show that it supports puppetv4

**Comment by Phil-Friderici** [comment]
Thanks everybody, released in v0.6.1

**Pull request merged**